quodlibet/quodlibet/ext/events/synchronize_to_device.py

```diff
@@ -0,0 +1,168 @@
+
+    def PluginPreferences(self, parent):
+        vbox = Gtk.VBox(spacing=6)
+
+        query_path = os.path.join(get_user_dir(), 'lists', 'queries.saved')
+        try:
+            with open(query_path, 'r', encoding="utf-8") as query_file:
+                log = Gtk.TextView()
+                log.set_left_margin(5)
+                log.set_right_margin(5)
+                log.editable = False
+                buffer = log.get_buffer()
+                scroll = Gtk.ScrolledWindow()
+                scroll.set_size_request(-1, 100)
+                scroll.add(log)
+
+                def append(text):
+                    GLib.idle_add(lambda: buffer.insert(buffer.get_end_iter(),
+                                                        text + "\n"))
+                    GLib.idle_add(
+                        lambda: log.scroll_to_mark(buffer.get_insert(), 0.0,
+                                                   True, 0.5, 0.5))
+
+                queries = {}
+                for query_string in query_file:
+                    name = next(query_file).strip()
+                    queries[name] = Query(query_string.strip())
+
+                for query_name, query in queries.items():
+                    check_button = ConfigCheckButton(query_name, "plugins",
+                                                     self._config_key(
+                                                         query_name))
+                    check_button.set_active(self.config_get_bool(query_name))
+                    vbox.pack_start(check_button, False, True, 0)
+
+                def synchronize():
+                    enabled_queries = []
+                    for query_name, query in queries.items():
+                        if self.config_get_bool(query_name):
+                            enabled_queries.append(query)
+
+                    selected_songs = []
+                    for song in app.library.itervalues():
+                        if any(query.search(song) for query in
+                               enabled_queries):
+                            selected_songs.append(song)
+
+                    filename_list = []
+                    destination_path = config.get('plugins', self.c_path)
+                    try:
+                        for song in selected_songs:
+                            if not self.running:
+                                append("Stopped the synchronization.")
+                                return
+
+                            # prevent the application from becoming unreponsive
+                            while Gtk.events_pending():
+                                Gtk.main_iteration()
+
+                            song_path = song['~filename']
+                            song_file_name = "{} - {} - {} - {}.mp3".format(
+                                song("genre"), song("artist"),
+                                song("album"), song("title"))
+                            valid_chars = "-_.() %s%s" % (
```

Comment:
Why is this here? Unicode characters should be allowed. This also generates duplicate filenames if the only difference in the filename is unicode characters.

Reply:
That definitely exists if you think about non-latin scripts like the ones used in Japanese, Chinese etc. Ideally there would be some fallback or option for this, but it should be okay for now.